File: 209.MinimumSizeSubarraySum.py
class Solution:
    # Sliding Window
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        index, currSum, minLength = 0, 0, float('inf')
        for i in range(len(nums)):
            currSum += nums[i]
            while currSum >= target:
                minLength = min(minLength, i-index+1)
                currSum -= nums[index]
                index += 1
        return 0 if minLength == float('inf') else minLength

File: 209.MinimumSizeSubarraySum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int curr = 0;
        int minLength = Integer.MAX_VALUE;
        int start = 0;
        for (int i=0; i<nums.length; i++) {
            curr += nums[i];
            while (curr >= target && start<nums.length) {
                minLength = Math.min(minLength, i-start+1);
                curr -= nums[start++];
            }
        }
        return minLength == Integer.MAX_VALUE ? 0 : minLength;
        
    }
}